core/src/main/java/tc/oc/pgm/filters/FlagStateFilter.java

```diff
@@ -23,6 +27,11 @@ public FlagStateFilter(
     this.state = state;
   }
 
+  @Override
+  public Collection<Class<? extends Event>> getRelevantEvents() {
+    return ImmutableList.of(FlagStateChangeEvent.class);
```

Comment:
For all of these one-event collections consider using a singleton:
```suggestion
    return Collections.singleton(FlagStateChangeEvent.class);
```
They may be simpler than an immutable list, and they're immutable too

Reply:
`ImmutableList.of(E element)` return a `SingetonImmutableList`, but the native Collection will always be less "fancy". Since we don't need any features of the `ImmutableList` ill go for `Collections.singleton`